src/main/java/com/artipie/api/RepositoryName.java

```diff
@@ -12,6 +12,7 @@
  *
  * @since 0.26
  */
+@SuppressWarnings("PMD.DataClass")
 public class RepositoryName {
```

Comment:
Sorry, I've missed the moment then this interface became a class. Public static methods are not allowed in Artipie, operator `new` in java is meant to create objects, not static methods. Here is how this should be done in object-thinking way: 
```
public interface RepositoryName {
    /**
     * Username path parameter name.
     */
    String UNAME = "uname";

    /**
     * Repository path parameter name.
     */
    String RNAME = "rname";

    /**
     * The name of the repository.
     * @return String name
     */
    String toString();

    /**
     * Repository name from request (from vertx {@link RoutingContext}). 
     * @since 0.26
     */
    class FromRequest implements RepositoryName {

        /**
         * Layout.
         */
        private final String layout;

        /**
         * Repository name.
         */
        private final RoutingContext context;

        /**
         * Ctor.
         *
         * @param context Context
         * @param layout  Layout
         */
        public FromRequest(final RoutingContext context, final String layout) {
            this.context = context;
            this.layout = layout;
        }

        /**
         * Provides string representation of repository name in toString() method.
         * <ul>
         *     <li>'reponame' for flat layout</li>
         *     <li>'username/reponame' for org layout</li>
         * </ul>
         *
         * @checkstyle NoJavadocForOverriddenMethodsCheck (10 lines)
         */
        @Override
        public String toString() {
            final String reponame;
            if (new Layout.Flat().toString().equals(this.layout)) {
                reponame = this.context.pathParam(RepositoryName.RNAME);
            } else {
                reponame = new Org(
                    this.context.pathParam(RepositoryName.UNAME),
                    this.context.pathParam(RepositoryName.RNAME)
                ).toString();
            }
            return reponame;
        }
    }

    /**
     * Repository name for flat layout.
     * @since 0.26
     */
    class Flat implements RepositoryName {

        /**
         * Repository name.
         */
        private final String name;

        /**
         * Ctor.
         * @param name Name
         */
        public Flat(final String name) {
            this.name = name;
        }

        @Override
        public String toString() {
            return this.name;
        }
    }

    /**
     * Repository name for org layout is combined from username and reponame:
     * 'username/reponame'.
     * @since 0.26
     */
    class Org implements RepositoryName {

        /**
         * Repository name.
         */
        private final String rname;

        /**
         * User name.
         */
        private final String uname;

        /**
         * Ctor.
         * @param rname Repository name
         * @param uname User name
         */
        public Org(final String rname, final String uname) {
            this.rname = rname;
            this.uname = uname;
        }

        @Override
        public String toString() {
            return String.format("%s/%s", this.uname, this.rname);
        }
    }

}
```
`FromReequest` implementation will be used in all verticles (http endpoints), the other two mostly in tests.
Validation for repository names can be extracted into separate class:
```

/**
 * Valid repository name.
 * @since 0.26
 */
public class ValidRepositoryName {

    /**
     * The name to test.
     */
    private final String name;

    /**
     * Ctor.
     * @param name The name to test
     */
    public ValidRepositoryName(final String name) {
        this.name = name;
    }

    /**
     * Ctor.
     * @param name The name to test
     */
    public ValidRepositoryName(final RepositoryName name) {
        this(name.toString());
    }

    /**
     * Checks if the repository name is valid. The name is considered valid if it does
     * not contain reserved words `_storages, _permissions, _credentials` in it.
     *
     * @return True is the name is valid
     */
    public boolean isValid() {
        return Stream.of("_storages", "_permissions", "_credentials")
            .filter(this.name::contains).findAny().isEmpty();
    }
}
```
it is very likely, that we will extend this class in the future to add some more checks.